src/MemoEditor.js

```diff
@@ -0,0 +1,34 @@
+import { useState, useEffect } from 'react';
+
+export default function MemoEditor({ selectedMemo, deleteMemo, saveMemo }) {
+  const [content, setContent] = useState(selectedMemo.content);
+
+  useEffect(() => {
+    setContent(selectedMemo.content);
+  }, [selectedMemo]);
+
+  const deleteSelectedMemo = () => {
+    deleteMemo(selectedMemo.id);
+  };
+
+  const saveClickHandler = () => {
+    saveMemo(selectedMemo.id, content);
+  };
+
+  const handleChange = (event) => {
+    setContent(event.target.value);
+  };
+
+  return (
+    <div>
+      <form>
+        <textarea rows="40" cols="60" value={content} onChange={handleChange} />
+        <br />
+        <button onClick={() => saveClickHandler(selectedMemo.id)}>保存</button>
```

Comment:
`saveClickHandler`は引数を取らないので、ここで渡している`selectedMemo.id`は無駄になってしまっていますね。（JavaScriptでは無駄な引数を渡してもエラーになりません）

なので、以下のようにした方が良いと思います。
```suggestion
        <button onClick={saveClickHandler}>保存</button>
```
また、React公式が推している命名として、こういうイベントリスナに登録する関数は`handleXXXX`という名前にしておくと分かりやすいです。`handleClickSave`とかそんな感じにしておくといいんじゃないかなと思います。

Reply:
ご指摘,説明ありがとうございます。
引数不要でした。
関数名をhandleClickSaveに変更致しました。